napari/components/layerlist.py

```diff
@@ -321,4 +321,6 @@ def _selection_context(self) -> dict:
     'same_shape': (
         lambda s: len({getattr(x.data, 'shape', ()) for x in s}) == 1
     ),
+    'active_layer_dtype': lambda s: s.active
+    and str(getattr(s.active.data, 'dtype', '')),
```

Comment:
Can we use `napari.utils._dtype.normalize_dtype` here? See https://github.com/napari/napari/pull/3380/files for example usage. This will allow use with tensorstore arrays. (Currently all dtypes are selected because the string doesn't match the NumPy string.)

I'll also say I'd prefer a real function here to this almost-too-clever one-liner. I find the construct `bool_expr and value` to be rather unreadable and obscure... (And @tlambert03 might remember that it bit us very confusingly with Pytest a while back.)

Reply:
Sounds good to me. Will update accordingly.

Reply:
@jni does the latest update make sense?